NuRadioReco/modules/iftElectricFieldReconstructor/iftElectricFieldReconstructor.py

```diff
@@ -443,36 +479,37 @@ def __prepare_traces(
+                    for sim_channel in station.get_sim_station().iter_channels():
+                        if sim_channel.get_id() == channel_id:
+                            if sim_channel_sum is None:
+                                sim_channel_sum = sim_channel
+                            else:
+                                sim_channel_sum += sim_channel
                 if sim_channel_sum is not None:
-                    sim_channel_sum.apply_time_shift(-channel.get_parameter(chp.signal_time_offset), True)
+                    sim_channel_sum.apply_time_shift(-self.__time_offsets[i_channel], True)
                     ax2_1.plot(sim_channel_sum.get_times(), sim_channel_sum.get_filtered_trace(passband, filter_type='butterabs') / units.mV, c='k', alpha=.5)
                     ax2_1.set_xlim([sim_channel_sum.get_trace_start_time() - 50, sim_channel_sum.get_times()[-1] + 50])
-                    sim_channel_sum.apply_time_shift(channel.get_parameter(chp.signal_time_offset), True)
+                    sim_channel_sum.apply_time_shift(self.__time_offsets[i_channel], True)
 
             channel.apply_time_shift(-toffset[np.argmax(correlation_sum)])
             self.__data_traces[i_channel] = channel.get_trace()[:self.__trace_samples]
             self.__noise_levels[i_channel] = np.sqrt(np.mean(channel.get_trace()[self.__trace_samples + 1:]**2))
-            self.__n_shifts[i_channel] = int((toffset[np.argmax(correlation_sum)] + time_offset) * channel.get_sampling_rate())
-            self.__trace_start_times[i_channel] = channel.get_trace_start_time() + (toffset[np.argmax(correlation_sum)] + time_offset)
+            self.__n_shifts[i_channel] = int((toffset[np.argmax(correlation_sum)] + self.__time_offsets[i_channel]) * channel.get_sampling_rate())
+            self.__trace_start_times[i_channel] = channel.get_trace_start_time() + (toffset[np.argmax(correlation_sum)] + self.__time_offsets[i_channel])
             if self.__debug:
                 ax2_2 = fig2.add_subplot(len(self.__used_channel_ids), 2, 2 * i_channel + 2)
                 ax2_2.grid()
                 ax2_2.plot(np.arange(len(self.__data_traces[i_channel])) / channel.get_sampling_rate(), self.__data_traces[i_channel])
-            channel.apply_time_shift(channel.get_parameter(chp.signal_time_offset) + toffset[np.argmax(correlation_sum)], True)
+            channel.apply_time_shift(self.__time_offsets[i_channel] + toffset[np.argmax(correlation_sum)], True)
         self.__scaling_factor = np.max(self.__data_traces)
         self.__data_traces /= self.__scaling_factor
         self.__noise_levels /= self.__scaling_factor
 
         if self.__debug:
             ax1_2.plot(correlation_sum)
             fig2.tight_layout()
-            fig2.savefig('{}_{}_traces.png'.format(event.get_run_number(), event.get_id()))
+            fig2.savefig('{}/{}_{}_traces.png'.format(self.__plot_folder, event.get_run_number(), event.get_id()))
```

Comment:
It would be nice if this debug plot were also properly labelled (in particular, I'm not sure what's shown on the right hand side)